db/dao/todo.go

```diff
@@ -0,0 +1,15 @@
+package dao
+
+import (
+	"time"
+
+	"gorm.io/gorm"
+)
+
+type Todo struct {
+	gorm.Model
+	Description string     `json:"description" gorm:"not null"`
+	CompletedAt *time.Time `json:"completed_at"`
+	Completed   bool       `json:"completed" gorm:"default:false"`
+	UserID      uint       `json:"user_id" gorm:"not null"`
```

Comment:
What is the purpose of `gorm` tag and what is the difference of adding or not adding `not null` and `default` here?